src/ecomail.ts

```diff
@@ -0,0 +1,40 @@
+import { Volunteer } from "./slack";
+import axios from "axios";
+
+interface Subscriber {
+  email: string;
+}
+
+interface SubscriberData {
+  subscriber_data: Subscriber[];
+  update_existing: boolean;
+}
+
+function buildSubscriberData(volunteers: Volunteer[]): SubscriberData {
+  const subs: Subscriber[] = volunteers
+    .filter(v => v.email != null)
+    .map(v => {
+      return {
+        email: v.email ?? ""
```

Comment:
Je tahle kontrola nutná, když se předtím filtruje na not-null email? Pokud je problém v ostatních `nullish` hodnotách, dal by se predikát filteru předělat na `v.email ?? false`.

Cílem je mít tady jednoduchý výraz `.map(v => {email: v.email}` (nevím, jestli tahle zkrácená syntaxe u objektu půjde).

Reply:
Tohle je kvůli kontrole typů, není dost chytrá na to, aby věděla, že po filtru na nenullové hodnoty už ten e-mail nemůže být null. Jde to nějak obejít? Taky mně to tam vadí :) Četl jsem třeba [tohle](https://stackoverflow.com/questions/43118692), ale rychlé řešení našeho případu mě z toho nenapadá.

Ta zkrácená syntaxe u objektů nefunguje, jak říkáš. Myslím, že už je toho v tu chvíli na parser moc :)

Reply:
Tak první dirty trick, co mě napadá je:
```
return {
        email: v.email
} as Subscriber
```

Další možnost by byla https://www.typescriptlang.org/docs/handbook/release-notes/typescript-2-0.html#non-null-assertion-operator (občas to bývá zakázané v eslintu nebo ts-configu).

Tedy 
```
return {
        email: v.email!
} 
```

Reply:
Aha, to druhé je přesně force-unwrap ze Swiftu, super, nevěděl jsem, že to existuje i tady. Podle mě je to správné řešení, když máš tu záruku hned o řádek výš.

Reply:
Souhlasím, jenom existují... názory :-) https://github.com/Microsoft/TypeScript/issues/9637